app/src/main/java/lvc/com/movies/ui/listmovies/ListMoviesViewModel.java

```diff
@@ -0,0 +1,83 @@
+package lvc.com.movies.ui.listmovies;
+
+import android.app.Application;
+import android.arch.lifecycle.AndroidViewModel;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Transformations;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+
+import lvc.com.movies.database.DatabaseFacade;
+import lvc.com.movies.database.MovieDao;
+import lvc.com.movies.database.MovieDatabase;
+import lvc.com.movies.model.Data;
+import lvc.com.movies.model.Movie;
+import lvc.com.movies.model.MovieSearch;
+import lvc.com.movies.repository.MoviesRepository;
+import lvc.com.movies.util.Objects;
+
+/**
+ * Created by leonardo2050 on 28/01/18.
+ */
+
+public class ListMoviesViewModel extends AndroidViewModel {
```

Comment:
What is the purpose of using `AndroidViewModel` here?

Reply:
ViewModel is lifecycle aware component to manage UI data. So we know that everything there will survive events like screen rotations.
AndroidViewModel is the same thing, but it also provides a Application reference in the constructor and I need it to instantiate my Database class. 
I believe I could avoid it if I had implemented a dependency injection strategy.

Reply:
To be honest, I've never used this strategy. I was curious about and it looks very interesting. Thanks!